src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/InterfaceDispatchMapIndirectionNode.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    class InterfaceDispatchMapIndirectionNode : EmbeddedObjectNode, ISymbolNode
+    {
+        TypeDesc _type;
+
+        public InterfaceDispatchMapIndirectionNode(TypeDesc type)
+        {
+            _type = type;
+            base.Offset = 1; // 1 is not a valid offset, so when the wrapper object emitter sets offsets, it will become more reasonable
+        }
+
+        public override bool StaticDependenciesAreComputed
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        string ISymbolNode.MangledName
+        {
+            get
+            {
+                if (base.Offset != 1)
```

Comment:
I'm not sure I follow why we have two names for a single symbol that changes depending on how far along are we in object emission.